exp/controllers/machinepool_controller_phases.go

```diff
@@ -108,6 +108,33 @@ func (r *MachinePoolReconciler) reconcileExternal(ctx context.Context, cluster *
 		return external.ReconcileOutput{Paused: true}, nil
 	}
 
+	if strings.HasSuffix(ref.Kind, external.TemplateSuffix) {
+		owner := &metav1.OwnerReference{
+			APIVersion: expv1.GroupVersion.String(),
+			Kind:       "MachinePool",
+			Name:       m.Name,
+			UID:        m.UID,
+		}
```

Comment:
Isn't the owner reference already set below?

Reply:
I don't t think so

Reply:
Current line 118, there is a call that sets it `controllerutil.SetControllerReference(m, obj, r.Client.Scheme())`

Reply:
I use this owner in line 123 passing owner as input in ColneTemplate func so i can't remove it.